dkg-gadget/src/testing.rs

```diff
@@ -0,0 +1,1210 @@
+	) -> Result<(), sp_blockchain::Error> {
+		todo!()
+	}
+	fn unpin_block(&self, _: <TestBlock as sp_api::BlockT>::Hash) {
+		todo!()
+	}
+
+	fn begin_operation(&self) -> sp_blockchain::Result<Self::BlockImportOperation> {
+		todo!()
+	}
+
+	fn begin_state_operation(
+		&self,
+		_operation: &mut Self::BlockImportOperation,
+		_block: <TestBlock as BlockT>::Hash,
+	) -> sp_blockchain::Result<()> {
+		todo!()
+	}
+
+	fn commit_operation(
+		&self,
+		_transaction: Self::BlockImportOperation,
+	) -> sp_blockchain::Result<()> {
+		todo!()
+	}
+
+	fn finalize_block(
+		&self,
+		_hash: <TestBlock as BlockT>::Hash,
+		_justification: Option<sp_runtime::Justification>,
+	) -> sp_blockchain::Result<()> {
+		todo!()
+	}
+
+	fn append_justification(
+		&self,
+		_hash: <TestBlock as BlockT>::Hash,
+		_justification: sp_runtime::Justification,
+	) -> sp_blockchain::Result<()> {
+		todo!()
+	}
+
+	fn blockchain(&self) -> &Self::Blockchain {
+		todo!()
+	}
+
+	fn usage_info(&self) -> Option<sc_client_api::UsageInfo> {
+		todo!()
+	}
+
+	fn offchain_storage(&self) -> Option<Self::OffchainStorage> {
+		Some(self.inner.offchain_storage.clone())
+	}
+
+	fn state_at(&self, _hash: <TestBlock as BlockT>::Hash) -> sp_blockchain::Result<Self::State> {
+		todo!()
+	}
+
+	fn revert(
+		&self,
+		_n: sp_api::NumberFor<TestBlock>,
+		_revert_finalized: bool,
+	) -> sp_blockchain::Result<(
+		sp_api::NumberFor<TestBlock>,
+		std::collections::HashSet<<TestBlock as BlockT>::Hash>,
+	)> {
+		todo!()
+	}
+
+	fn remove_leaf_block(&self, _hash: <TestBlock as BlockT>::Hash) -> sp_blockchain::Result<()> {
+		todo!()
+	}
+
+	fn get_import_lock(&self) -> &RwLock<()> {
+		todo!()
+	}
+
+	fn requires_full_sync(&self) -> bool {
+		todo!()
+	}
+}
+
+impl HeaderBackend<TestBlock> for TestBackend {
+	fn header(&self, _id: H256) -> sp_blockchain::Result<Option<<TestBlock as BlockT>::Header>> {
+		todo!()
+	}
+
+	fn info(&self) -> sp_blockchain::Info<TestBlock> {
+		todo!()
+	}
+
+	fn status(&self, _id: H256) -> sp_blockchain::Result<sp_blockchain::BlockStatus> {
+		todo!()
+	}
+
+	fn number(
+		&self,
+		_hash: <TestBlock as BlockT>::Hash,
+	) -> sp_blockchain::Result<Option<<<TestBlock as BlockT>::Header as sp_api::HeaderT>::Number>> {
+		todo!()
+	}
+
+	fn hash(
+		&self,
+		_number: sp_api::NumberFor<TestBlock>,
+	) -> sp_blockchain::Result<Option<<TestBlock as BlockT>::Hash>> {
+		todo!()
+	}
+}
+
+impl ProvideRuntimeApi<TestBlock> for TestBackend {
+	type Api = DummyApi;
+	fn runtime_api(&self) -> sp_api::ApiRef<Self::Api> {
+		sp_api::ApiRef::from(self.inner.api.clone())
+	}
+}
+impl AuxStore for TestBackend {
+	fn insert_aux<
+		'a,
+		'b: 'a,
+		'c: 'a,
+		I: IntoIterator<Item = &'a (&'c [u8], &'c [u8])>,
+		D: IntoIterator<Item = &'a &'b [u8]>,
+	>(
+		&self,
+		_insert: I,
+		_delete: D,
+	) -> sp_blockchain::Result<()> {
+		todo!()
+	}
+
+	fn get_aux(&self, _key: &[u8]) -> sp_blockchain::Result<Option<Vec<u8>>> {
+		todo!()
+	}
+}
+
+#[derive(Clone)]
+pub struct DummyApi {
```

Comment:
Can this be broken out into another file? Can this testing system be broken out into a new crate? This file is like a mish-mosh of a lot of unimplemented structs/traits with poor documentation.

Reply:
I've moved this into the test-orchestrator crate for now, that way it's no longer inside `dkg_gadget`